saltgui/static/scripts/routes/home.js

```diff
@@ -75,6 +90,51 @@ class HomeRoute extends Route {
     if(this.keysLoaded && this.jobsLoaded) this.resolvePromise();
   }
 
+  _addMenu(element) {
+    var menuDropdown = this._createDiv("run-command-button", "");
+    var menuButton = this._createDiv("menu-dropdown", "&#9658;");
+    menuDropdown.appendChild(menuButton);
+    var menuDropdownContent = this._createDiv("menu-dropdown-content", "");
+    //menuDropdownContent.appendChild(this._createDiv("run-command-button", "x1"));
+    //menuDropdownContent.appendChild(this._createDiv("run-command-button", "x2"));
+    //menuDropdownContent.appendChild(this._createDiv("run-command-button", "x3"));
+    menuDropdown.appendChild(menuDropdownContent);
+    element.appendChild(menuDropdown);
+    return menuDropdownContent;
+  }
```

Comment:
Ok, I really like the menu popup dropdown, it makes much more sense like this! Personally I would make the font a bit smaller so it fits a bit better with the rest:

<img width="923" alt="menu" src="https://user-images.githubusercontent.com/121401/40966477-e311da6e-68b0-11e8-9c66-12cc4906a998.png">

As for the code, I would suggest to create a new class for the menu and create an instance of it in the constructor (like this: https://github.com/maerteijn/SaltGUI/blob/master/saltgui/static/scripts/router.js#L4), so you have the menu code separated from the rest (otherwise we will create one big monolithic class which will be hard to maintain)

Reply:
The font enlargement has been removed

Reply:
The dropdownmenu system has been refactored into a separate class